### PR TITLE
feature/webinfo? feature/persistence? feature/mongo?

### PR DESCRIPTION
**Pull request by Nanabell**
I don't even know what is all contained in this pr anymore. 
What I did:
- [x] Remove Quarterdeck and replace it with Mongo

Supposedly this branch also contains the web info command and handles queue persistence on shutdown.
Have fun.

### CODE REVIEW

File: FredBoat/src/main/java/fredboat/util/ratelimit/Blacklist.java [hunk 1]
```diff
@@ -82,13 +82,13 @@ public boolean isBlacklisted(long id) {
         //first of all, ppl that can never get blacklisted no matter what
         if (userWhiteList.contains(id)) return false;
 
-        BlacklistEntry blEntry = blacklistService.fetchBlacklistEntry(id);
+        BlacklistEntity blEntry = repository.fetch(id).block();
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/util/ratelimit/Blacklist.java]: `.block()` is evil!

The name should probably also be changed

File: FredBoat/src/main/java/fredboat/util/ratelimit/Blacklist.java [hunk 2]
```diff
@@ -102,7 +102,7 @@ public boolean isBlacklisted(long id) {
     public long hitRateLimit(long id) {
         //update blacklist entry of this id
         long blacklistingLength = 0;
-        BlacklistEntry blEntry = blacklistService.fetchBlacklistEntry(id);
+        BlacklistEntity blEntry = repository.fetch(id).block();
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/util/ratelimit/Blacklist.java]: See other comment

File: config/templates/fredboat.yml [hunk 3]
```diff
@@ -54,6 +54,7 @@ config:
                                   # things happening to your bot in the selfhosting chat you will be publicly taunted.
   prefix:            '<<'         # Default prefix used by the bot
   botAdmins:         []           # Add comma separated userIds and roleIds that should have access to bot admin commands. Find role ids with the ;;roleinfo command
+  botOwners:         []           # Add comma separated userIds that should have access to bot OWNER commands. Mainly intended for bots that belong to teams
   autoBlacklist:     true         # Set to true to automatically blacklist users who frequently hit the rate limits
   game:              ""           # Set the displayed game/status. Leave empty quote marks for the default status
   continuePlayback:  false        # Set to true to force the player to continue playback even if left alone
```
Comment (Frederikam) [inline config/templates/fredboat.yml]: Please remove the quarterdeck info in this file

File: FredBoat/src/main/java/fredboat/db/api/BaseRepository.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+    fun fetch(id: ID): Mono<T>
+
+    fun update(mono: Mono<T>): Mono<T>
+
+    fun update(target: T): Mono<T>
+
+    fun remove(id: ID): Mono<Void>
+
+    fun default(id: ID): T?
```
[the remark this answers is not in the document]
Comment (Nanabell) [reply FredBoat/src/main/java/fredboat/db/api/BaseRepository.kt]: This is nullable on purpose if null is returned it is assumed that this entity cannot have a default value. See SearchResultRepositoryImpl as an example

[the remark this answers is not in the document]
Comment (Frederikam) [reply FredBoat/src/main/java/fredboat/db/api/BaseRepository.kt]: I know, but that still seems like bad practice. You could create a new interface that has this method, which extends this one.

File: FredBoat/src/main/java/fredboat/command/config/PrefixCommand.kt [hunk 5]
```diff
@@ -61,13 +63,9 @@ class PrefixCommand(cacheMetrics: CacheMetricsCollector, name: String, vararg al
...
                 .concurrencyLevel(Launcher.botController.appConfig.shardCount)  //each shard has a thread (main JDA thread) accessing this cache many times
-                .build(CacheLoader.asyncReloading(CacheLoader.from<Long, Optional<String>> {
-                    guildId -> Launcher.botController.prefixService.getPrefix(Prefix.GuildBotId(
-                        guildId!!,
-                        botId
-                ))
-                },
-                        Launcher.botController.executor))!!
+                .build(CacheLoader.asyncReloading(CacheLoader.from<Long, Optional<String>> { guildId ->
+                    Optional.ofNullable(Launcher.botController.guildSettingsRepository.fetch(guildId!!).block()?.prefix)
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/command/config/PrefixCommand.kt]: Don't block here. You might not be able to get CacheBuilder to use an async loader

File: FredBoat/src/main/java/fredboat/feature/I18n.java [hunk 6]
```diff
@@ -72,10 +74,7 @@ public static FredBoatLocale getLocale(@Nonnull Guild guild) {
     @Nonnull
     public static FredBoatLocale getLocale(long guild) {
         try {
-            return LANGS.getOrDefault(getBotController().getGuildConfigService().fetchGuildConfig(guild).getLang(), DEFAULT);
-        } catch (DatabaseNotReadyException e) {
-            //don't log spam the full exceptions or logs
-            return DEFAULT;
+            return LANGS.getOrDefault(getBotController().getGuildSettingsRepository().fetch(guild).block(Duration.ofSeconds(10)).getLang(), DEFAULT);
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/feature/I18n.java]: Don't use block

File: FredBoat/src/main/java/fredboat/util/rest/TrackSearcher.java [hunk 7]
```diff
@@ -180,10 +174,22 @@ public AudioPlaylist searchForTracks(String query, long cacheMaxAge, int timeout
     @Nullable
     private AudioPlaylist fromCache(SearchProvider provider, String searchTerm, long cacheMaxAge) {
         try {
-            SearchResult.SearchResultId id = new SearchResult.SearchResultId(provider, searchTerm);
-            return searchResultService.getSearchResult(id, cacheMaxAge)
-                    .map(searchResult -> searchResult.getSearchResult(audioPlayerManager))
-                    .orElse(null);
+            SearchResultId id = new SearchResultId(provider, searchTerm);
+            SearchResult result = repository.fetch(id).block();
```
Comment (Frederikam) [inline FredBoat/src/main/java/fredboat/util/rest/TrackSearcher.java]: Don't block here. This method could perhaps be migrated to Kotlin and be made suspending

**Pull request merged**